rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYaml.java

```diff
@@ -110,9 +111,11 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
             @Override
             public Yaml.Document visitDocument(Yaml.Document document, ExecutionContext ctx) {
                 if ("$".equals(key)) {
-                    return document.withBlock((Yaml.Block) new MergeYamlVisitor<>(document.getBlock(), yaml,
-                            Boolean.TRUE.equals(acceptTheirs), objectIdentifyingProperty).visitNonNull(document.getBlock(),
-                            ctx, getCursor()));
+                    Yaml.Document d = document.withBlock((Yaml.Block)
+                            new MergeYamlVisitor<>(document.getBlock(), yaml, Boolean.TRUE.equals(acceptTheirs), objectIdentifyingProperty)
+                                    .visitNonNull(document.getBlock(), ctx, getCursor())
+                    );
+                    return getCursor().getMessage(REMOVE_PREFIX, false) ? d.withEnd(d.getEnd().withPrefix("")) : d;
```

Comment:
I actually don't like very much that I needed to put this here; I would rather keep the cursor messages locked in the `MergeYamlVisitor` class. I couldn't make it work though, because the ` visitDocument` will not be called in that class. 

If I overlooked something please let me know! It would be really nice to let this MergeYaml visitor file untouched.